### PR TITLE
WIP: notification page

### PR DESCRIPTION
**Pull request by Natalie90206**
Added dot component and profile photo component in the notification page.

### CODE REVIEW

File: app/src/main/res/drawable/shape_profile_photo.xml [hunk 1]
```diff
@@ -0,0 +1,8 @@
+<?xml version="1.0" encoding="utf-8"?>
+<shape xmlns:android="http://schemas.android.com/apk/res/android"
```
Comment (jingweihu) [inline app/src/main/res/drawable/shape_profile_photo.xml:2]: removed

File: app/src/main/java/com/laiteam/echowall/component/DotSpec.java [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+        return Column.create(c)
+                .paddingDip(YogaEdge.HORIZONTAL, 8f)
+                .paddingDip(YogaEdge.VERTICAL, 4f)
+                .child(getComponent(c))
+                .build();
+    }
+
+    static Component getComponent(ComponentContext c) {
+        return Image.create(c)
+                .drawableRes(R.drawable.shape_dot).foregroundColor(Color.parseColor("#F5B844"))
```
Comment (jingweihu) [inline app/src/main/java/com/laiteam/echowall/component/DotSpec.java:30]: size and color should be props

File: app/src/main/java/com/laiteam/echowall/component/ProfilePhotoSpec.java [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+
+@LayoutSpec
+public class ProfilePhotoSpec {
+    @OnCreateLayout
+    static Component onCreateLayout(ComponentContext c, @Prop String imageUrl) {
+
+        return FrescoImage.create(c)
+                .controller(getImageComponent(imageUrl))
+                .failureImageRes(R.drawable.shape_profile_photo)
+                .widthDip(45).heightDip(45)
```
Comment (jingweihu) [inline app/src/main/java/com/laiteam/echowall/component/ProfilePhotoSpec.java:26]: Size should be props as well

**Review by jingweihu — CHANGES_REQUESTED** [review]

File: app/src/main/java/com/laiteam/echowall/component/DotSpec.java [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+import com.laiteam.echowall.R;
+
+
+@LayoutSpec
+public class DotSpec {
+    @OnCreateLayout
+    static Component onCreateLayout(ComponentContext c) {
+
+        return Column.create(c)
+                .paddingDip(YogaEdge.HORIZONTAL, 8f)
```
Comment (jingweihu) [inline app/src/main/java/com/laiteam/echowall/component/DotSpec.java:22]: dot itself shouldn't have any padding

File: app/src/main/java/com/laiteam/echowall/component/ProfilePhotoSpec.java (hunk 3 above)
Comment (jingweihu) [reply app/src/main/java/com/laiteam/echowall/component/ProfilePhotoSpec.java:26]: screenshots ?

File: app/src/main/java/com/laiteam/echowall/component/ProfilePhotoSpec.java [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+import com.laiteam.echowall.R;
+
+@LayoutSpec
+public class ProfilePhotoSpec {
+    @OnCreateLayout
+    static Component onCreateLayout(ComponentContext c, @Prop String imageUrl) {
+
+        return FrescoImage.create(c)
+                .controller(getImageComponent(imageUrl))
+                .failureImageRes(R.drawable.shape_profile_photo)
```
Comment (jingweihu) [inline app/src/main/java/com/laiteam/echowall/component/ProfilePhotoSpec.java:25]: failureRes should be optional props 2

**Review by jingweihu — CHANGES_REQUESTED** [review]

File: app/src/main/java/com/laiteam/echowall/component/ProfilePhotoSpec.java (hunk 3 above)
Comment (Natalie90206) [reply app/src/main/java/com/laiteam/echowall/component/ProfilePhotoSpec.java:26]: Screen Shot 2021-03-28 at 9 34 18 PM

File: app/src/main/java/com/laiteam/echowall/component/DotSpec.java (hunk 2 above)
Comment (Natalie90206) [reply app/src/main/java/com/laiteam/echowall/component/DotSpec.java]: updated

File: app/src/main/java/com/laiteam/echowall/component/DotSpec.java (hunk 4 above)
Comment (Natalie90206) [reply app/src/main/java/com/laiteam/echowall/component/DotSpec.java]: removed paddings in Dot component

File: app/src/main/java/com/laiteam/echowall/component/ProfilePhotoSpec.java (hunk 5 above)
Comment (Natalie90206) [reply app/src/main/java/com/laiteam/echowall/component/ProfilePhotoSpec.java:26]: **valid url, load successfully:**
Screen Shot 2021-03-28 at 9 34 18 PM

**invalid url, load failure:**
Screen Shot 2021-03-28 at 9 46 08 PM

**Pull request closed without merging**